goss.yaml

```
@@ -100,34 +104,62 @@ process:
     running: true
 
 command:
-  "aws --version":
+  aws --version:
     exit-status: 0
 
-  "git --version":
+  git --version:
     exit-status: 0
 
-  "git-lfs --version":
+  git-lfs --version:
     exit-status: 0
 
-  "/etc/cron.hourly/docker-low-disk-gc":
+  systemctl is-enabled docker-gc.timer:
     exit-status: 0
 
-  "/etc/cron.hourly/docker-gc":
+  /usr/local/bin/docker-gc:
     exit-status: 0
 
-  # Checks that docker is running
-  "docker info":
+  systemctl is-enabled docker-low-disk-gc.timer:
     exit-status: 0
-    timeout: 30000 # it can take some time for the daemon to start
 
-  # Checks that docker containers can run
-  "docker run --rm -v /var/run/docker.sock:/var/run/docker.sock docker:latest version":
+  /usr/local/bin/docker-low-disk-gc:
     exit-status: 0
-    timeout: 30000 # it can take some time to download the image
 
-  # Checks that permissions
-  'sh -c "docker run --rm -v \"$PWD:/pwd\" alpine:latest touch /pwd/test && stat -c %U:%G test"':
+  # Check docker userns is enabled
+  docker info --format='{{ `{{range .SecurityOptions}}{{if eq . "name=userns"}}{{"success"}}{{end}}{{end}}` }}':
```

Weirdly, on my machine, the nested ``{{ `{{ range`` constructions don't work.  How about something like this, perhaps a bit more intuitive?

```suggestion
  # Check docker userns is enabled
  docker info --format='{{range .SecurityOptions}}{{ if eq . "name=cgroupns" }}{{.}}{{end}}{{end}}':
```

---

So the nested templates are necessary because goss treats its yaml file as a template, so it would try to evaluate `.SecurityOptions` in the `--format` argument when it is meant to pass it through to docker.

I'll add a comment about this.

---

Ah gotcha, so you're saying my suggestion works in my Terminal but wouldn't work in the GOSS yaml file?  Thanks for the explanation 👌

---

Yeah, and mine has the opposite issue.

I could put the whole line in a nested template, that might be the best of both words.